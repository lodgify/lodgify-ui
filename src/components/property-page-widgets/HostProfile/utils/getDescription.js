import React from 'react';
import { size } from 'lodash';

import { GridColumn } from 'layout/GridColumn';
import { Paragraph } from 'typography/Paragraph';

/**
 * @typedef {Object} null
 * @param {string} description
 * @return {Object|null}
 */
export const getDescription = description =>
  size(description) > 0 ? (
    <GridColumn computer={7} mobile={12} tablet={12}>
      <Paragraph size="medium">{description}</Paragraph>
    </GridColumn>
  ) : null;
