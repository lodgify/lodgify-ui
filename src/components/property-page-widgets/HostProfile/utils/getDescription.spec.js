import { getDescription } from './getDescription';

describe('`getDescription`', () => {
  it('should have the right structure', () => {
    const description = 'lorem upsum dulur sumut';
    const actual = getDescription(description);

    expect(actual).toMatchSnapshot();
  });
  it('should return null if the description length is 0 or undefined', () => {
    const description = undefined;
    const actual = getDescription(description);

    expect(actual).toBe(null);
  });
});
