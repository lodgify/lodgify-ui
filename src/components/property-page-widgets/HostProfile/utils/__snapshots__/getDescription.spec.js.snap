// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`getDescription\` should have the right structure 1`] = `
<GridColumn
  computer={7}
  mobile={12}
  tablet={12}
  verticalAlignContent="top"
  width={null}
>
  <Paragraph
    size="medium"
    weight={null}
  >
    lorem upsum dulur sumut
  </Paragraph>
</GridColumn>
`;
